src/containers/shared/components/TxToken.tsx

```diff
@@ -18,12 +21,28 @@ function getTokenPair(
     type === 'Payment'
   ) {
     const first =
-      amount?.amount && amount.amount !== fee ? amount.currency : undefined
+      amount?.amount && amount.amount !== fee ? (
+        <Currency currency={amount.currency} />
+      ) : undefined
     const second =
-      amount2?.amount && amount2.amount !== fee ? amount2.currency : undefined
+      amount2?.amount && amount2.amount !== fee ? (
+        <Currency currency={amount2.currency} />
+      ) : undefined
 
     if (first && second) {
-      return first + (type === 'Payment' ? ' for ' : ' and ') + second
+      return (
+        <Trans
+          i18nKey={
+            type === 'Payment'
+              ? 'token_for_transaction_swap'
+              : 'token_for_transaction'
```

Comment:
I'm not really sure what `token_for_transaction` means - is there a more descriptive label that could be used?